kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapGeneralLedgerServiceImpl.java

```diff
@@ -373,4 +376,124 @@ public void setPurapUseTaxEntryArchiveService(PurapUseTaxEntryArchiveService pur
+                if (item.getItemQuantity() != null) {
+                    item.setItemOutstandingEncumberedQuantity(item.getItemQuantity().subtract(item.getItemInvoicedTotalQuantity()));
+                }
+                else {
+                    // if order qty is null, outstanding encumbered qty should be null
+                    item.setItemOutstandingEncumberedQuantity(null);
+                }
+
+                // Set amount
+                if (item.getItemOutstandingEncumberedQuantity() != null) {
+                    //do math as big decimal as doing it as a KualiDecimal will cause the item price to round to 2 digits
+                    KualiDecimal itemEncumber = new KualiDecimal(item.getItemOutstandingEncumberedQuantity().bigDecimalValue().multiply(item.getItemUnitPrice()));
+
+                    //add tax for encumbrance
+                    KualiDecimal itemTaxAmount = item.getItemTaxAmount() == null ? ZERO : item.getItemTaxAmount();
+                    itemEncumber = itemEncumber.add(itemTaxAmount);
+
+                    item.setItemOutstandingEncumberedAmount(itemEncumber);
+                }
+                else {
+                    if (item.getItemUnitPrice() != null) {
+                        item.setItemOutstandingEncumberedAmount(new KualiDecimal(item.getItemUnitPrice().subtract(item.getItemInvoicedTotalAmount().bigDecimalValue())));
+                    }
+                }
+                //This section was modified for UAF-1156/UAF-1164
+                for (Iterator<PurApAccountingLine> iter = item.getSourceAccountingLines().iterator(); iter.hasNext();) {
+                    PurchaseOrderAccount account = (PurchaseOrderAccount) iter.next();
+                    account.setItemAccountOutstandingEncumbranceAmount(account.getAmount());
+                    account.setAlternateAmountForGLEntryCreation(account.getItemAccountOutstandingEncumbranceAmount());
+                }
+                //end of modified section
+            }
+        }
+
+        PurchaseOrderDocument oldPO = purchaseOrderService.getCurrentPurchaseOrder(po.getPurapDocumentIdentifier());
+
+        if (oldPO == null) {
+            throw new IllegalArgumentException("Current Purchase Order not found - poId = " + oldPO.getPurapDocumentIdentifier());
+        }
+
+        List<SourceAccountingLine> newAccounts = purapAccountingService.generateSummaryWithNoZeroTotalsUsingAlternateAmount(po.getItemsActiveOnly());
+        List<SourceAccountingLine> oldAccounts = purapAccountingService.generateSummaryWithNoZeroTotalsUsingAlternateAmount(oldPO.getItemsActiveOnlySetupAlternateAmount());
+
+        Map<SourceAccountingLine, KualiDecimal> combination = new HashMap<SourceAccountingLine, KualiDecimal>();
+
+        // Add amounts from the new PO
+        for (Iterator<SourceAccountingLine> iter = newAccounts.iterator(); iter.hasNext();) {
+            SourceAccountingLine newAccount = (SourceAccountingLine) iter.next();
```

Comment:
Unnecessary cast here, iter is typed already in this version of java and the cast is a remnant of not having generesized Iterator. (Not worth re-doing just for the PR just for this, but if there ends up being an additional reason, might as well knock it out.)